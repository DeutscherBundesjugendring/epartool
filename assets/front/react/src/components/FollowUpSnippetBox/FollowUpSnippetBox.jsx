import React from 'react';
import Reputation from '../Reputation/Reputation';


const FollowUpSnippetBox = (props) => {
  let glyphClasses = 'followup-type-icon glyphicon';
  let isType = false;
  let glyphTitle = '';

  if (props.type === 'supporting') {
    glyphClasses += ' glyphicon-heart';
    glyphTitle = props.typeSupportingLabel;
    isType = true;
  } else if (props.type === 'action') {
    glyphClasses += ' glyphicon-play';
    glyphTitle = props.typeActionLabel;
    isType = true;
  } else if (props.type === 'rejected') {
    glyphClasses += ' glyphicon-minus-sign';
    glyphTitle = props.typeRejectedLabel;
    isType = true;
  } else if (props.type === 'end') {
    glyphClasses += ' glyphicon-lock';
    glyphTitle = props.typeEndLabel;
    isType = true;
  }

  return (
    <div
      className="
        well
        well-bordered
        followup-flow
        followup-well
        followup-well-collapsible
        followup-well-link
        followup-type-wrap
        followup-type-wrap-inner
      "
    >
      <div className="js-followup-box-head">
        {isType && (
          <div className="followup-type followup-type-right followup-type-right-alt">
            <span className={glyphClasses} aria-hidden="true" />
            <span className="followup-type-title">{glyphTitle}</span>
          </div>
        )}
        <img
          src={props.document.previewImageLink}
          alt={props.document.title}
          width="120"
          className="offset-bottom img-responsive"
        />
      </div>
      <div
        className="js-followup-box-content"
        // eslint-disable-next-line react/no-danger
        dangerouslySetInnerHTML={{ __html: props.snippetExplanation }}
      />
      <div className="followup-reputation">
        <Reputation
          dislikeAction={props.dislikeAction}
          dislikeCount={props.dislikeCount}
          dislikeLabel={props.dislikeLabel}
          likeAction={props.likeAction}
          likeCount={props.likeCount}
          likeLabel={props.likeLabel}
          votingLimitError={props.votingLimitError}
        />
      </div>
    </div>
  );
};

FollowUpSnippetBox.propTypes = {
  type: React.PropTypes.oneOf(['general', 'supporting', 'action', 'rejected', 'end']).isRequired,
  snippetExplanation: React.PropTypes.string.isRequired,
  likeAction: React.PropTypes.func.isRequired,
  likeCount: React.PropTypes.number.isRequired,
  likeLabel: React.PropTypes.string.isRequired,
  dislikeAction: React.PropTypes.func.isRequired,
  dislikeCount: React.PropTypes.number.isRequired,
  dislikeLabel: React.PropTypes.string.isRequired,
  document: React.PropTypes.shape({
    previewImageLink: React.PropTypes.string.isRequired,
    title: React.PropTypes.string.isRequired,
  }).isRequired,
  typeActionLabel: React.PropTypes.string.isRequired,
  typeEndLabel: React.PropTypes.string.isRequired,
  typeRejectedLabel: React.PropTypes.string.isRequired,
  typeSupportingLabel: React.PropTypes.string.isRequired,
  votingLimitError: React.PropTypes.string.isRequired,
};

export default FollowUpSnippetBox;
