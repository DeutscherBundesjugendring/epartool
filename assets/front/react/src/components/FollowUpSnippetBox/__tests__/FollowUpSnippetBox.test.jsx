import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import FollowUpSnippetBox from '../FollowUpSnippetBox';


injectTapEventPlugin();

const element = (
  <FollowUpSnippetBox
    snippetExplanation="Snippet"
    likeAction={() => {}}
    likeCount={0}
    dislikeAction={() => {}}
    dislikeCount={0}
    followPathAction={() => {}}
    followPathLabel="Follow path"
  />
);

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(
      React.cloneElement(element)
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls like action', () => {
    const spy = sinon.spy();
    const component = shallow(
      React.cloneElement(element, { likeAction: spy })
    );

    component.find('ThumbButton').first().simulate('touchTap', { preventDefault: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls dislike action', () => {
    const spy = sinon.spy();
    const component = shallow(
      React.cloneElement(element, { dislikeAction: spy })
    );

    component.find('ThumbButton').at(1).simulate('touchTap', { preventDefault: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls follow path action', () => {
    const spy = sinon.spy();
    const component = shallow(
      React.cloneElement(element, { followPathAction: spy })
    );

    component.find('RaisedButton').first().simulate('touchTap', { preventDefault: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });
});
