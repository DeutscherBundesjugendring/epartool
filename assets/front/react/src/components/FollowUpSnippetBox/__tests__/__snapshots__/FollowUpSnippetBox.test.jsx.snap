// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="well well-bordered followup-well followup-well-link"
>
  <EmbeddedVideo
    height={120}
    videoId="youtube-video-id"
    videoService="youtube"
    width={240}
  />
  <p>
    Snippet
  </p>
  <div
    className="offset-bottom-small"
  >
    <span
      className="badge"
    >
      0
    </span>
    <ThumbButton
      disabled={false}
      onTouchTap={[Function]}
      type="like"
    />
    <span
      className="badge"
    >
      0
    </span>
    <ThumbButton
      disabled={false}
      onTouchTap={[Function]}
      type="dislike"
    />
  </div>
  <RaisedButton
    disabled={false}
    label="Follow path"
    onTouchTap={[Function]}
  />
</div>
`;

exports[`rendering renders correctly without optional props 1`] = `
<div
  className="well well-bordered followup-well followup-well-link"
>
  <p>
    Snippet
  </p>
  <div
    className="offset-bottom-small"
  >
    <span
      className="badge"
    >
      0
    </span>
    <ThumbButton
      disabled={false}
      onTouchTap={[Function]}
      type="like"
    />
    <span
      className="badge"
    >
      0
    </span>
    <ThumbButton
      disabled={false}
      onTouchTap={[Function]}
      type="dislike"
    />
  </div>
  <RaisedButton
    disabled={false}
    label="Follow path"
    onTouchTap={[Function]}
  />
</div>
`;
